mainmatter/introduction/graphics_hardware/index.tex

```diff
@@ -142,29 +142,62 @@ \subsubsection{AMD}
+Embedded \glspl{GPU} are also connected to the rest of the Intel \gls{CPU} through this ring.
+The on-die ring interconnect bus between the \gls{CPU} cores, caches, and embedded \gls{GPU}. 
+Each connected component has its own interface for how it should use the ring (additionally, each \gls{CPU} core is considered as its own unique component). 
+The SoC ring interconnect is a bi-directional ring which has a 32-byte wide data bus.  
+When the \gls{CPU} needs to interact with off-chip systems, this is facilitated by the ring as well. 
+
+The architecture of Intel Processor Graphics was designed to be modular in order to make it more scalable. 
+This means it is possible to design Intel Graphics HD units that will work on other devices than desktop computers. 
+The architecture consists of compute components called execution units, execution units are clustered into groups called subslices, and subslices are clustered into slices. 
+Each component will be further explained.
+
+\fig{figures/Intel_ExecutionUnit}{Architecture of a execution unit, figure taken from \cite{computegen9}}{IntelEU}{1} 
+
+\ref{IntelEU} shows the fundamental building block of Generation 8 and 9 \gls{GPU} architecture, the execution unit. 
+The figure shows that each Execution Unit contains 7 threads, and each thread contains 128 SIMD 32-bit registers.  
+
+On each cycle an Execution Unit can issue up to four different instructions which are sourced from up to four different threads, the thread arbiter dispatches the instructions for execution.
+Each Execution Unit has a pair of FPUs used for computation (They support both Integer and Floatin-point computation despite being called FPUs).
+
+\fig{figures/Intel_SubSlice}{Architecture of a Subslice, figure taken from \cite{computegen9}}{IntelSS}{1}
+In this architecture(same for both gen8 and 9), arrays of Execution Units in groups are called a subslice. A subslice contains 8 Execution Units. 
+This means each subslice can run up to 56 threads simultaneously. 
+each subslice has its own local thread dispatcher, which is used to partition threads into the execution units in a subslice.
+The sampler is a read only memory fetch unit, used for sampling textures and images.
+The dataport is used to memory operations.
+
+\fig{figures/Intel_Slice}{Architecture of a Slice, figure taken from \cite{computegen9}}{IntelS}{1}
+Subslices grouped together are called slices. For most Generation 9 Intel Graphics HD \glspl{CPU} they are grouped as three.
+This means a gen8 and 9 with a slice containing three subslices can run at most 168 threads simultaneously.
+Besides grouping subslices together, a slice adds additional logic for thread dispatching and a level 3 cache.
+
+\fig{figures/Intel_Complete}{Putting each component in its place, figure taken from \cite{computegen9}}{IntelC}{1}
+It is worth to note that an intel graphics HD \gls{CPU} may contain more than one slice to give a higher throughput.
```

Comment:
"intel" with lower case i? check for consistency (and correctness)...

Reply:
done